src/cmd/services/m3coordinator/downsample/options.go

```diff
@@ -637,6 +1004,76 @@ func (o DownsamplerOptions) newAggregatorFlushManagerAndHandler(
 	return flushManager, handler
 }
 
+// Force the local aggregator client to implement client.Client.
+var _ client.AdminClient = (*aggregatorLocalAdminClient)(nil)
+
+type aggregatorLocalAdminClient struct {
+	agg aggregator.Aggregator
+}
+
+func newAggregatorLocalAdminClient() *aggregatorLocalAdminClient {
+	return &aggregatorLocalAdminClient{}
+}
+
+func (c *aggregatorLocalAdminClient) SetAggregator(agg aggregator.Aggregator) {
+	c.agg = agg
+}
```

Comment:
I unexported it but kept it since `	// NB: Can't do this at construction time since needs to be passed as an
	// option to the aggregator constructor.`